server/datastore/mysql/hosts_test.go

```diff
@@ -4248,70 +4248,111 @@ func testOSVersions(t *testing.T, ds *Datastore) {
 	require.NoError(t, err)
 
 	// all hosts
-	osVersions, err := ds.OSVersions(ctx, nil, nil, nil)
+	osVersions, err := ds.OSVersions(ctx, nil, nil, nil, nil)
 	require.NoError(t, err)
 
 	require.True(t, time.Now().After(osVersions.CountsUpdatedAt))
 	expected := []fleet.OSVersion{
-		{HostsCount: 1, Name: "CentOS 8.0.0", NameOnly: "CentOS", Version: "8.0.0", Platform: "rhel", ID: osByNameVers["CentOS 8.0.0"].ID},
-		{HostsCount: 2, Name: "Ubuntu 20.4.0", NameOnly: "Ubuntu", Version: "20.4.0", Platform: "ubuntu", ID: osByNameVers["Ubuntu 20.4.0"].ID},
-		{HostsCount: 1, Name: "macOS 12.1.0", NameOnly: "macOS", Version: "12.1.0", Platform: "darwin", ID: osByNameVers["macOS 12.1.0"].ID},
-		{HostsCount: 3, Name: "macOS 12.2.1", NameOnly: "macOS", Version: "12.2.1", Platform: "darwin", ID: osByNameVers["macOS 12.2.1"].ID},
-		{HostsCount: 3, Name: "macOS 12.3.0", NameOnly: "macOS", Version: "12.3.0", Platform: "darwin", ID: osByNameVers["macOS 12.3.0"].ID},
+		{HostsCount: 1, Name: "CentOS 8.0.0", NameOnly: "CentOS", Version: "8.0.0", Platform: "rhel"},
```

Comment:
Just curious, so moving forward, the plan is to deprecate `Name` in favor of `NameOnly` and `Version` ?

Reply:
Not quite, whenever we do breaking changes to the API, I believe product wants to migrate `name` to just include the name without the version then drop `name_only` field from the API response.

Reply:
Sounds good - thanks

Reply:
As mentioned on the slack conversation - I think we should look into refactoring the vulnerability code to use the `operating_system` table instead.